src/super_gradients/training/kd_model/kd_model.py

```diff
@@ -18,101 +19,194 @@ class KDModel(SgModel):
+        :param teacher_arch_params: (dict) Architecture H.P. e.g.: block, num_blocks, num_classes, etc for teacher
+
+        """
+        self._validate_subnet_num_classes(student_arch_params)
+        self._validate_subnet_num_classes(teacher_arch_params)
+        if teacher_arch_params['num_classes'] != student_arch_params['num_classes']:
+            raise ValueError("num_classes inconsistent between student and teacher architecture params")
+
+    def _validate_subnet_num_classes(self, subnet_arch_params):
+        """
+        Derives num_classes in student_arch_params/teacher_arch_params from dataset interface or raises an error
+         when none is given
+
+        @param subnet_arch_params: Arch params for student/teacher
+
+        """
+
+        if 'num_classes' not in subnet_arch_params.keys():
+            if self.dataset_interface is None:
+                raise Exception('Error', 'Number of classes must be defined in students and teachers arch params or '
+                                         'by connecting to a dataset interface')
+            else:
+                subnet_arch_params['num_classes'] = len(self.classes)
+
+    @staticmethod
+    def instantiate_net(architecture: Union[KDModule, KDModule.__class__, str], arch_params: dict,
+                        checkpoint_params: dict, *args, **kwargs) -> tuple:
+        """
+        Instantiates kd_module according to architecture and arch_params, handles pretrained weights for the student
+         and teacher networks, and the required module manipulation (i.e head replacement) for the teacher network.
+
+        :param architecture: String, KDModule or uninstantiated KDModule class describing the netowrks architecture.
+        :param arch_params: Architecture's parameters passed to networks c'tor.
+        :param checkpoint_params: checkpoint loading related parameters dictionary with 'pretrained_weights' key,
+            s.t it's value is a string describing the dataset of the pretrained weights (for example "imagenent").
+
+        :return: instantiated netowrk i.e KDModule, architecture_class (will be none when architecture is not str)
+        """
+
+        student_architecture = get_param(kwargs, "student_architecture")
+        teacher_architecture = get_param(kwargs, "teacher_architecture")
+        student_arch_params = get_param(kwargs, "student_arch_params")
+        teacher_arch_params = get_param(kwargs, "teacher_arch_params")
+        student_arch_params = core_utils.HpmStruct(**student_arch_params)
+        teacher_arch_params = core_utils.HpmStruct(**teacher_arch_params)
+        student_pretrained_weights = get_param(checkpoint_params, 'student_pretrained_weights')
+        teacher_pretrained_weights = get_param(checkpoint_params, 'teacher_pretrained_weights')
+
+        student, _ = SgModel.instantiate_net(student_architecture, student_arch_params,
```

Comment:
@shaydeci  - I think that the IDE's recommendation to define instantiate_net as a staticmethod is wrong since it breaks the inheritance...

If we remove the "@staticmethod" from SGModel.instantiate_net than we can simply replace 
SgModel.instantiate_net(student_architecture, student_arch_params,
                                             {"pretrained_weights": student_pretrained_weights}) 
with 
super().instantiate_net(student_architecture, student_arch_params,
                                             {"pretrained_weights": student_pretrained_weights})

Which is the behavior we are after...


WDYT?